### PR TITLE
test: Several test cases are supported for reporting

### PR DESCRIPTION
**Pull request by utam0k**
## Description

This report style comes from [our RFC](https://www.notion.so/gitpod/How-to-create-the-Quality-Assurance-Report-for-the-workspace-a8b7ee2570c44a25aed7bd7d4766f1b1)

For now, I will adopt it for some test cases, and if we like, I will create PRs to handle others as well.

## Related Issue(s)

NONE

## How to test

```console
 ./run.sh -r report.csv -s workspace 
cat report.csv
```

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=workspace
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by utam0k** [comment]
For some reason, `.werft/jobs/build/trigger-integration-tests.ts` of my change doesn't affect :sob: 
@mads-hartmann Probably you know the reason?

**Comment by utam0k** [comment]
This PR have passed the integration test
https://werft.gitpod-dev.com/job/gitpod-custom-to-test-report.3

File: test/pkg/report/report.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package report
+
+import "testing"
+
+func SetupReport(t *testing.T, typ Feature, desc string) {
```
Comment (jenting) [inline test/pkg/report/report.go:9]:
```suggestion
func SetupReport(t *testing.T, feat Feature, desc string) {
```

Comment (utam0k) [reply test/pkg/report/report.go:9]: Good English. I love it. Thanks

Comment (utam0k) [reply test/pkg/report/report.go]: fexed

**Comment by utam0k** [comment]
@jenting PTAL

File: test/pkg/report/features.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package report
+
+type Feature string
+
+const (
+	FeatureNone          Feature = "none"
```
Comment (jenting) [inline test/pkg/report/features.go:10]: Probably we don't require this one. We should try to find a feature sets that the integration tests belong to.

Comment (utam0k) [reply test/pkg/report/features.go]: Make sense. Deleted it

**Review by jenting — APPROVED** [review]
LGTM

**Comment by mads-hartmann** [comment]
>For some reason, .werft/jobs/build/trigger-integration-tests.ts of my change doesn't affect 😭
@mads-hartmann Probably you know the reason?

@utam0k Did you resolve this issue? If not, can you describe a bit more what you're seeing and how it is different from what you expected? ☺️

**Comment by kylos101** [comment]
@utam0k added to our project :smile:

**Comment by utam0k** [comment]
> @utam0k added to our project smile

Thanks!

**Comment by utam0k** [comment]
> @utam0k Did you resolve this issue? If not, can you describe a bit more what you're seeing and how it is different from what you expected? relaxed

@mads-hartmann Sorry for my lack of explanation. I changed `.werft/jobs/build/trigger-integration-tests.ts` but it seems that this change didn't reflect. My changes did not seem to be reflected when triggered by the PR checkbox. I expected that when I checked the checkbox on this PR, I could run new `werft/jobs/build/trigger-integration-tests.ts`

**Comment by mads-hartmann** [comment]
@utam0k This is due to job protection, any job triggered from GitHub will use job-protection while anything triggered from the CLI won't ([see note about it here](https://gitpod.slack.com/archives/C01KPEPSK2N/p1662548038447179)). This is confusing unfortunately. A good rule of thumb is to always use the Werft CLI to trigger jobs if you have changed anything in `.werft` ☺️

**Pull request merged**